edx_sga/sga.py

```diff
@@ -705,6 +808,32 @@ def file_storage_path(self, sha1, filename):
         )
         return path
 
+    def get_user(self):
+        """returns user object"""
+        try:
+            user = self.runtime.get_real_user(self.xmodule_runtime.anonymous_student_id)
+        except Exception:  # pylint: disable=broad-except
+            user = None
+
+        return user
```

Comment:
Please change this function to `get_required_user` or something like that, do not catch the exception, and remove the various `if user:` conditionals. This function is used in 2 places, and unless I'm missing something, it is called where it would not make any sense for there to be no actual user. If there is no user, that is truly exceptional, and therefore it should raise an exception and the request should fail. It's not good practice and it's potentially dangerous for us to let the app work normally and use default values if something truly unexpected is happening